### PR TITLE
Support lang and file filters in chat

### PR DESCRIPTION
**Pull request by anastasiya1155**

### CODE REVIEW

**Review by ggordonhall — APPROVED** [review]

**Review by nerdypepper — APPROVED** [review]
code changes lgtm.

File: server/bleep/src/agent/exchange.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@ use rand::seq::SliceRandom;
 pub struct Exchange {
     pub id: uuid::Uuid,
     pub query: SemanticQuery<'static>,
+    pub raw_query: String,
```
[the remark this answers is not in the document]
Comment (rsdy) [reply server/bleep/src/agent/exchange.rs:15]: We could, but it's easier for the frontend to pick it from here.

[the remark this answers is not in the document]
Comment (anastasiya1155) [reply server/bleep/src/agent/exchange.rs:15]: doesn't matter much for the FE, can change if you want

[the remark this answers is not in the document]
Comment (rsdy) [reply server/bleep/src/agent/exchange.rs:15]: @anastasiya1155 I have made this change now

[the remark this answers is not in the document]
Comment (anastasiya1155) [reply server/bleep/src/agent/exchange.rs]: updated the FE

**Pull request merged**